### PR TITLE
Bigger example _and_ fix render logic

### PR DESCRIPTION
**Pull request by juliangruber**
screen shot 2017-05-23 at 09 50 25

so far everything works fine!

### CODE REVIEW

**Comment by juliangruber** [comment]
next i want to add reordering

**Comment by juliangruber** [comment]
ok, this triggers a bug now: when you move down the first element, the placeholder element will be rendered

**Comment by juliangruber** [comment]
this is a comparison of the dom: https://gist.github.com/juliangruber/9f76bea837968fd4d27560eeaf56c067

**Comment by juliangruber** [comment]
It's a bit tricky to explain what was going on here, but I'll try my best:

Before, we were overwriting `this._render` in this module. Nanocomponent calls to `this._render()` and _then_ afterwards applies the `this._ID` logic. That means however, that the new tree that is being morphed in this module, doesn't have the `ID` yet! => nanomorph thinks it's different nodes

I therefore changed it to be the other way around. In this module, we set `this._render` to the handler function passed to `.on('render', handler)`, and then override __`this.render`__ with our morphing logic. So before it was nanocomponent calling microcomponent's render function, now it's microcomponent calling nanocomponent's render function. Which is the right way around.

The example now works 100% for me, as long as you update your local nanocomponent to this PR: https://github.com/yoshuawuyts/nanocomponent/pull/33

File: index.js [hunk 1]
```diff
@@ -34,15 +34,21 @@ Microcomponent.prototype.on = function (eventname, handler) {
   assert.equal(typeof handler, 'function', 'microcomponent.on handler should be type function')
 
   if (eventname === 'render') {
-    this._render = function (props) {
+    this._render = function () {
```
Comment (yoshuawuyts) [inline index.js:37]: can't we shorthand this to:
```js
var render = function () {
  return handler.call(this)
}
```

Or perhaps shorter if we skip declaring it here, and on line 47 we do:
```js
handler.call(this, props)
```
Or am I missing something?

Comment (yoshuawuyts) [reply index.js:37]: Oh, re-read your comments. This is err, woah quite strange haha. Ah well, we can always clean it up later

**Comment by juliangruber** [comment]
3.1.4

**Pull request merged**